src/orange/OrangeTrackView.hh

```diff
@@ -76,7 +76,7 @@ class OrangeTrackView
     inline CELER_FUNCTION Propagation find_next_step(real_type max_step);
 
     // Find the distance to the nearest boundary in any direction
-    inline CELER_FUNCTION real_type find_safety();
+    inline CELER_FUNCTION real_type find_safety() const;
```

Comment:
Since find safety might cache the value or use other temporary storage I intentionally made this non const.

Reply:
So, did you also leave GeoTrackView* instead of const GeoTrackView& during the refactoring? If so, no problem to drop this MR without merging - just was trying to apply consistent rules.

Reply:
Yeah, because we've been following a convention of "passing as pointer rather than reference to indicate that it will be modified", I left it as a pointer intentionally. It's ugly I know. I'm totally fine with directly passing `safety` and `direction` into the MscScatter if you're good with that. The only reason I didn't is that `find_safety` was guarded by `if (rho > params_.geom_limit)`, so with your suggested changes we're *always* finding the safety instead of just *sometimes*. I have no idea if that'll affect performance on GPU and/or CPU but if it simplifies the interface and the code I think it's OK.

Reply:
Got it!  So, let's leave codes as they are, as we will likely cache safety information (the last safety along with safety_origin?) which will definitely reduce the number of direct calls to vecgeom/orange, but simply update safety information at the current position as a short cut in certain cases (especially within a relatively large fiducial volume of calorimeter-like detectors).  I will close this MR if you agree.

Reply:
All right, sounds good. Thanks for trying to improve the interface though!